src/services/services.ts

```diff
@@ -260,26 +260,30 @@ namespace ts {
         }
 
         public getFirstToken(sourceFile?: SourceFile): Node {
-            let children = this.getChildren();
-            for (let child of children) {
-                if (child.kind < SyntaxKind.FirstNode) {
-                    return child;
-                }
+            let children = this.getChildren(); // why isn't sourceFile passed as an argument??
+            if (!(children && children.length > 0)) { return undefined; }
+
+            let child = children[0];
 
-                return child.getFirstToken(sourceFile);
+            if (child.kind < SyntaxKind.FirstNode) {
+                return child;
             }
+
+            return child.getFirstToken(sourceFile);
         }
 
         public getLastToken(sourceFile?: SourceFile): Node {
             let children = this.getChildren(sourceFile);
-            for (let i = children.length - 1; i >= 0; i--) {
-                let child = children[i];
-                if (child.kind < SyntaxKind.FirstNode) {
-                    return child;
-                }
+            if (!children) { return undefined; }

```

Comment:
`getLastOrUndefined()` checks whether the array is empty and returns the last value or undefined accordingly.